e2e/rook-test-framework/transport/k8s_transport_client.go

```diff
@@ -0,0 +1,77 @@
+package transport
+
+import (
+
+	"github.com/rook/rook/e2e/rook-test-framework/utils"
+	"github.com/rook/rook/e2e/rook-test-framework/objects"
+)
+
+type k8sTransportClient struct {
+}
+
+func CreateNewk8sTransportClient() *k8sTransportClient {
+	return &k8sTransportClient{}
+}
+
+const (
+	defaultFailedCode = 1
+	kubectl_executable = "kubectl"
+)
+
+func (k *k8sTransportClient) ExecuteCmd(cmd []string) (stdout string, stderr string, err error) {
+	return utils.ExecuteCmdAndLogToConsole(kubectl_executable, cmd, []string{})
+
+}
+
+func (k *k8sTransportClient) Apply(cmdArgs []string) (stdout string, stderr string, err error) {
+	initialArgs := []string{"replace", "--force", "-f"}
+	cmdArgs = append(initialArgs, cmdArgs...)
+	return utils.ExecuteCmdAndLogToConsole("kubectl", cmdArgs, []string{})
+}
+
+func (k *k8sTransportClient) Execute(cmdArgs []string, optional []string) (stdout string, stderr string, exitCode int) {
+
+	if optional != nil {
+		if len(optional) == 1 {
+			initialArgs := []string{"exec", optional[0], "--"}
+			cmdArgs = append(initialArgs, cmdArgs...)
+		} else if len(optional) == 2 {
+			initialArgs := []string{"exec", "-n", optional[1], optional[0], "--"}
```

Comment:
that's odd that the optional args would be used out of order here instead of in the order they were given in.